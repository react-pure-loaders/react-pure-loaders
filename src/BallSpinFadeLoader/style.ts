import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';
import delay from '../utilities';
import { Grid } from '../types';

const radius = '25px';

const ballSpinFade = keyframes({
  '100%': {
    opacity: 1,
    transform: 'scale(1)',
  },
  '50%': {
    opacity: 0.3,
    transform: 'scale(0.4)',

  },
});

function ballSpinFadeLoader() {
  const n = 8;
  let i = 1;
  const grid: Grid = {};
  for (; i < 9; i += 1) {
    const quarter = (parseInt(radius, 10) / 2) + (parseInt(radius, 10) / 5.5);

    if (i === 1) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpinFade} 1s ${delay('0.12s', n, i - 1)} infinite linear`,
        left: 0,
        top: radius,
      };
    } else if (i === 2) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpinFade} 1s ${delay('0.12s', n, i - 1)} infinite linear`,
        left: quarter,
        top: quarter,
      };
    } else if (i === 3) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpinFade} 1s ${delay('0.12s', n, i - 1)} infinite linear`,
        left: radius,
        top: 0,
      };
    } else if (i === 4) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpinFade} 1s ${delay('0.12s', n, i - 1)} infinite linear`,
        left: quarter,
        top: `-${quarter}px`,
      };
    } else if (i === 5) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpinFade} 1s ${delay('0.12s', n, i - 1)} infinite linear`,
        left: 0,
        top: `-${radius}`,
      };
    } else if (i === 6) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpinFade} 1s ${delay('0.12s', n, i - 1)} infinite linear`,
        left: `-${quarter}px`,
        top: `-${quarter}px`,
      };
    } else if (i === 7) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpinFade} 1s ${delay('0.12s', n, i - 1)} infinite linear`,
        left: `-${radius}`,
        top: 0,
      };
    } else if (i === 8) {
      grid[`> div:nth-of-type(${i})`] = {
        animation: `${ballSpinFade} 1s ${delay('0.12s', n, i - 1)} infinite linear`,
        left: `-${quarter}px`,
        top: quarter,
      };
    }
  }

  return grid;
}

export default (color?: string) => ({
  ...ballSpinFadeLoader(),
  left: '-10px',
  position: 'relative' as 'relative',
  top: '-10px',
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    position: 'absolute' as 'absolute',
  },
});
