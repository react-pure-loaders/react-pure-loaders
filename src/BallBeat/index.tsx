/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
