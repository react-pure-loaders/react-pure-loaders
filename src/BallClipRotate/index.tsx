/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
