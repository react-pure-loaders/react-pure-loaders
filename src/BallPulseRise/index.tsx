/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
