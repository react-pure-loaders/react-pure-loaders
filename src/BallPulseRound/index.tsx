/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
