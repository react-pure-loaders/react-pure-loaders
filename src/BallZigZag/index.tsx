/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
