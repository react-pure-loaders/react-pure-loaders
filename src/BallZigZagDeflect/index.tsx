/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
