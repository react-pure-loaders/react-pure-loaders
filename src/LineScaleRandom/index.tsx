/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
