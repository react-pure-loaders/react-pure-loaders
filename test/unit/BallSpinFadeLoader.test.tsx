import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallSpinFadeLoader from '../../src/BallSpinFadeLoader';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallSpinFadeLoader>', () => {
    afterEach(cleanup);

    test('BallSpinFadeLoader should match snapshot', () => {
        const { container } = render(<BallSpinFadeLoader loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('BallSpinFadeLoader should have default color', () => {
        const { container } = render(<BallSpinFadeLoader loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('BallSpinFadeLoader should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallSpinFadeLoader color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallSpinFadeLoader should have no children', () => {
        const { container } = render(<BallSpinFadeLoader loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallSpinFadeLoader should have three children', () => {
        const { container } = render(<BallSpinFadeLoader loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(9);
    });
});
