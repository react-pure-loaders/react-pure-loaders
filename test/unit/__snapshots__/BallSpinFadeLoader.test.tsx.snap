// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallSpinFadeLoader> BallSpinFadeLoader should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    opacity: 0.3;
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    opacity: 0.3;
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    opacity: 0.3;
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    opacity: 0.3;
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    opacity: 0.3;
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    opacity: 0.3;
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    opacity: 0.3;
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }
}

@keyframes animation-0 {
  100% {
    opacity: 1;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  50% {
    opacity: 0.3;
    -webkit-transform: scale(0.4);
    -moz-transform: scale(0.4);
    -ms-transform: scale(0.4);
    transform: scale(0.4);
  }
}

.emotion-0 {
  left: -10px;
  position: relative;
  top: -10px;
}

.emotion-0>div:nth-of-type(1) {
  -webkit-animation: animation-0 1s -0.96s infinite linear;
  animation: animation-0 1s -0.96s infinite linear;
  left: 0;
  top: 25px;
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation: animation-0 1s -0.84s infinite linear;
  animation: animation-0 1s -0.84s infinite linear;
  left: 17.045454545454547px;
  top: 17.045454545454547px;
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation: animation-0 1s -0.72s infinite linear;
  animation: animation-0 1s -0.72s infinite linear;
  left: 25px;
  top: 0;
}

.emotion-0>div:nth-of-type(4) {
  -webkit-animation: animation-0 1s -0.60s infinite linear;
  animation: animation-0 1s -0.60s infinite linear;
  left: 17.045454545454547px;
  top: -17.045454545454547px;
}

.emotion-0>div:nth-of-type(5) {
  -webkit-animation: animation-0 1s -0.48s infinite linear;
  animation: animation-0 1s -0.48s infinite linear;
  left: 0;
  top: -25px;
}

.emotion-0>div:nth-of-type(6) {
  -webkit-animation: animation-0 1s -0.36s infinite linear;
  animation: animation-0 1s -0.36s infinite linear;
  left: -17.045454545454547px;
  top: -17.045454545454547px;
}

.emotion-0>div:nth-of-type(7) {
  -webkit-animation: animation-0 1s -0.24s infinite linear;
  animation: animation-0 1s -0.24s infinite linear;
  left: -25px;
  top: 0;
}

.emotion-0>div:nth-of-type(8) {
  -webkit-animation: animation-0 1s -0.12s infinite linear;
  animation: animation-0 1s -0.12s infinite linear;
  left: -17.045454545454547px;
  top: 17.045454545454547px;
}

.emotion-0>div {
  border-radius: 100%;
  height: 15px;
  margin: 2px;
  width: 15px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  position: absolute;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
